### ISSUE TITLE
v4.0 crash report - SQLiteException: no such table: worktag

### ISSUE THREAD

**Issue by misaochan** [commons-app/apps-android-commons#5014]
From google groups forum. Not sure if this is related to #5006 , it is a different table.

```
USER_COMMENT=I tried to upload a file
APP_VERSION_CODE=1027
APP_VERSION_NAME=4.0.1
ANDROID_VERSION=12
PHONE_MODEL=SM-A515F
STACK_TRACE=android.database.sqlite.SQLiteException: no such table: worktag (code 1 SQLITE_ERROR[1]): , while compiling: SELECT DISTINCT tag FROM worktag WHERE work_spec_id=?
at android.database.sqlite.SQLiteConnection.nativePrepareStatement(Native Method)
at android.database.sqlite.SQLiteConnection.acquirePreparedStatement(SQLiteConnection.java:1478)
at android.database.sqlite.SQLiteConnection.prepare(SQLiteConnection.java:916)
at android.database.sqlite.SQLiteSession.prepare(SQLiteSession.java:590)
at android.database.sqlite.SQLiteProgram.<init>(SQLiteProgram.java:63)
at android.database.sqlite.SQLiteQuery.<init>(SQLiteQuery.java:37)
at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:46)
at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:2088)
at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:2063)
at androidx.sqlite.db.framework.FrameworkSQLiteDatabase.query(FrameworkSQLiteDatabase.java:161)
at androidx.room.RoomDatabase.query(RoomDatabase.java:446)
at androidx.room.util.DBUtil.query(DBUtil.java:83)
at androidx.work.impl.model.WorkTagDao_Impl.getTagsForWorkSpecId(WorkTagDao_Impl.java:94)
at androidx.work.impl.WorkerWrapper.run(WorkerWrapper.java:134)
at androidx.work.impl.utils.SerialExecutor$Task.run(SerialExecutor.java:91)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
at java.lang.Thread.run(Thread.java:920)

```

**Comment by misaochan** [comment]
I had more information from the user after talking to them:

> The upload failed three times:
> * first try resulted in this crash report
> I logged out and in again on the app.
> * again the app crashed and I sent the second crash report.
> I uploaded a different image, which succeeded.
> * tried to upload the first image again. This time the app did not crash, but upload did fail
> * retried the upload in the app and succeeded.
> 
> This was the image in question:
> MCDC perfect square  
> https://commons.wikimedia.org/wiki/File%3AMCDC_perfect_square_.jpg

This crash appears to be new in v4 - I looked through the v3 crashes for the last 60 days on the play store console and it did not exist there. I understand that it's not directly caused by us, but perhaps it might be worth looking at what changed, and if there is any workaround we can do?

**Comment by misaochan** [comment]
I will proceed to release without this, as we cannot reproduce this bug and the person who reported it did not get back to me regarding my question (re: reinstalling).

**Comment by misaochan** [comment]
This crash happened to me today while testing the app before release - it happened as soon as I submitted the upload after the final license step. The uploads before and after it were fine. In my case, the only difference I noticed was that the problematic upload did not have any coordinates, and I chose "skip" when prompted to choose coords for it.

I will proceed with the release regardless, but it would be good if anyone could take a look at this for future releases.
